pdns/pdns_recursor.cc

```diff
@@ -1218,7 +1218,14 @@ static void sendNODLookup(const DNSName& dname)
     // Send a DNS A query to <domain>.g_nodLookupDomain
     static const QType qt(QType::A);
     static const uint16_t qc(QClass::IN);
-    DNSName qname = dname + g_nodLookupDomain;
+    DNSName qname;
+    try {
+      qname = dname + g_nodLookupDomain;
+    }
+    catch(const std::range_error &e) {
```

Comment:
Any other exception that could happen here?

Reply:
Unsure - perhaps we could do a separate catchall that does log, so we'll find out some day.

Reply:
DNSName can also throw `out_of_range` and `runtime_error`, but not in the code paths we are triggering here